src/tracker.ts

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (c) 2020-2021 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+
+'use strict';
+
+import { RNSnowplowTracker } from './native';
+import { spLog } from './constants';
+import {
+  validateContexts,
+  validateSelfDesc,
+  validateScreenView,
+  validateStructured,
+  validatePageView,
+  validateTiming,
+  validateConsentGranted,
+  validateConsentWithdrawn,
+  validateEcommerceTransaction,
+} from './events';
+import type {
+  SelfDescribing,
+  EventContext,
+  ScreenViewProps,
+  StructuredProps,
+  PageViewProps,
+  TimingProps,
+  ConsentGrantedProps,
+  ConsentWithdrawnProps,
+  EcommerceTransactionProps,
+} from './types';
+
+/**
+ * Tracks a self-describing event
+ *
+ * @param ns {string} - the tracker namespace
+ * @param argmap {Object} - the event data
+ * @param ctxt {Array}- the event contexts
+ * @returns {Promise}
+ */
+function trackSelfDescribingEvent(
+  ns: string,
+  argmap: SelfDescribing,
+  ctxt: EventContext[] = []
```

Comment:
Where did all the vowels go? 
I think I'd extend `ns` to `namespace` too. No need to be short here, longer names are always more descriptive and easier to work with.